CORE-4996 consensual ledger structure

vlajos opened the pull request:
Continued from here: https://github.com/corda/corda-runtime-os/pull/1416

dimosr committed:
NOTICK - Create gonepostal branch

dimosr committed:
NOTICK - Reverse order to remove gonepostal from prefix

yift-r3 committed:
CORE-4661: Add help to p2p-setup configuration commands (#1168)

Add the help to the configurations command and 'show at file' to the other commands.

yift-r3 committed:
CORE-4660: Add exception handler to p2p-setup (#1167)

* Add Exception handler to the p2p-setup tool.
* Set required options as required.
* Adds support to print the stack tace in the p2p setup and fake ca.

yift-r3 committed:
NO-TICKET: Bump tag to the latest Gone Postal tag (#1171)

williamvigorr3 committed:
NOTICK Make TlsCertificatesPublisher read from the correct topic (#1207)

This probably means the LinkManager can't read already published TLS certificates (if it is restarted). This only causes duplicated entries to be published (as we read these from the hosting map).

yift-r3 committed:
CORE-4602: Create test tool to dump a topic into a file (#1198)

This PR adds a test tool to dump all the messages from a Kafka topic into a file.

dimosr committed:
NOTICK - Update artifact versions for gone postal branch (#1240)

Update artifact versions for gone postal branch

Commit:
NOTICK: apply open sourcing changes to gone postal branch (#1245)

* NOTICK: apply changes from open sourcing

Commit:
NOTICK: cherry pick from release/os/5.0 fix for base image credentals (#1252)

* NOTICK: cherry pick from release/os/5.0 fix for base image creds

dimosr committed:
NOTICK - Disable publishing of helm charts (#1311)

Disabling publishing of helm charts in the gone postal branch to avoid them being picked up from other jobs (e.g. soak testing) that want to use the latest version produced from the main branch.

dimosr committed:
NOTICK - switch gonepostal infix to p2p-preview (#1338)

Switching from `gonepostal` to `p2p-preview` to prevent confusion.

Requires: https://github.com/corda/corda-api/pull/378

Commit:
CORE-4939 use unique nexus ID for gone postal branch  (#1356)

* CORE-4939 use unique nexus ID

barrylapthorn committed:
CORE-4877 Extra persistence test comments (#1370)

lankydan committed:
NOTICK Update flow worker setup readme (#1373)

Add details of Link Manager setup which the flow worker now uses.

lankydan committed:
CORE-3764 Use dummy link manager temporarily (#1378)

The `LinkManagerProcessor` has not been integrated into the `FlowWorker`
application correctly which is affecting the overall network deployment.
Fall back to the dummy implementation until the issues have been
resolved.

nikinagy committed:
CORE-4757 - Publish locally hosted identities during static registration (#1355)

* publish locally hosted identities during static registration

* address comments

* remove unnecessary signing from test

alexey-kadyrov-r3 committed:
CORE-648 TCK (#1377)

kyriathar-r3 committed:
CORE-4999 Config Kafka/DB Reconciliation Writer (#1376)

CORE-4999 Config Kafka/DB Reconciliation Writer (#1376)

* Introduces config publish service API to match ReconcilerWriter API and wires it into config write service.

* Updates startup logic of config write service and DBProcessorImpl.

williamvigorr3 committed:
NOTICK Merge remote-tracking branch 'origin/release/gonepostal' into WillV/NOTICK-Merge-release-gone-postal-into-release-os-5

 Conflicts:
	applications/tools/p2p-test/p2p-setup/src/main/kotlin/net/corda/p2p/setup/Command.kt
	buildSrc/build.gradle
	buildSrc/src/main/groovy/DeployableContainerBuilder.groovy
	buildSrc/src/main/groovy/corda.common-publishing.gradle
	components/crypto/crypto-messaging-persistence-impl/build.gradle
	components/link-manager/build.gradle
	components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/TlsCertificatesPublisher.kt
	gradle.properties
	libs/configuration/configuration-publish-impl/build.gradle
	libs/configuration/configuration-publish/build.gradle
	libs/configuration/configuration-schema/messaging-configuration/build.gradle
	libs/messaging/db-messaging-impl/build.gradle
	libs/messaging/topic/build.gradle
	testing/bundles/testing-cats/build.gradle
	testing/bundles/testing-dogs/build.gradle

Sagar-Nenwani-R3 committed:
CORE-4867 Delete mgm-test modules (#1383)

Commit:
CORE-4864 - Refactor CPK structure to be able to query efficiently. (#1379)

* Change structure of the CPI/CPK metadata entities so that we can eagerly fetch everything in one query when using findAllCpiMetadata.
* Also simplified the CPI "Reset" code to not need the deletes.

yift-r3 committed:
CORE-4619: CSR creation API (#1316)

* CORE-4619: CSR creation API

* CORE-4619: Apply review comments

* Move x500 name to the path

* Change back to POST, remove email and usage, add context map and role

* Add signature spec

* Apply review comments

* Change name

charlieR3 committed:
Merge pull request #1384 from corda/WillV/NOTICK-Merge-release-gone-postal-into-release-os-5

NOTICK Merge release gone postal into release os 5

alexey-kadyrov-r3 committed:
CORE-648 increased test time as it fails on windows (#1386)

chrisr3 committed:
NOTICK: Close underlying SandboxGroupContext inside FlowSandboxGroupContext. (#1375)

benmcmahon100 committed:
NOTICK: Add labels to bootstrap jobs (#1351)

kyriathar-r3 committed:
CORE-5034 Fix DB/Kafka Reconciliation Writer API (#1391)

* Fix `ReconcilerWriter` and Kafka writers `put` and `remove` methods' apis
* Remove configuration dto and use configuration avro instead
* Update KDoc

vlajos committed:
CORE-5001 Move PrivacySalt to ledger:common-impl

Commit:
CORE-5032 - fix for uplolad cpi with shared cpk (#1389)

* Use merge when persisting CPI and CPK as the CPI may contain CPKs that already exist in the DB.
* Use OPTIMISTIC_FORCE_INCREMENT to ensure cpi entity version is bumped even if the CPI is identical.

fowlerrr committed:
CORE-3775: Update flow mapper and tests for correct lifecycle management (#1369)

* CORE-3775: Update flow mapper, tests, and documentation for correct lifecycle management

alexey-kadyrov-r3 committed:
CORE-5013 getting connection id from vnode info to access crypto-node database (#1393)

nikinagy committed:
CORE-4563 - Create group id if not defined (#1394)

* create group id if not defined

* address comment

nikinagy committed:
CORE-4830 - Assign soft HSM by default during static registration (#1381)

* assign soft hsms to member

* fix detekt

* address comments: add unit test

* create topic for hosted-identities to clear out error messages on member worker

alexey-kadyrov-r3 committed:
CORE-5047 separated EMF cache from the keys (#1399)

benmcmahon100 committed:
CORE-4596: Change to using topic bootstrapping done via the cli (#1270)

* Change to using topic bootstrapping done via the cli
* Fix classpath tweaking and fixup topic job
* Add some testing for output of topic creation
* Add config flag for SSL support
* Add resource requests to topic job
* Implement blocks to speed up bootstrapping
* Larger heap in order to fix SSL

YashNabar committed:
CORE-4535: Add MGM member info to member lookup cache (#1380)

kyriathar-r3 committed:
CORE-5014 Config Kafka/DB Reconciliation Readers (#1395) (#1395)

* Makes `ConfigurationReadService` a config reconciler Kafka reader.

* Introduces config reconciler DB reader.

* Abstracts out common code for DB readers.

* Adds is deleted flag on config entity for soft DB deletes.

lankydan committed:
CORE-5045 Register custom internal AMQP serializers in flow worker (#1397)

Call `AMQPSerializationScheme.registerCustomSerializers` in
`FlowSandboxServiceImpl` as it was incorrectly trying to scan for
internal serializers in `net.corda.internal.serialization.amqp.custom`
which are not currently OSGi components and `ThrowableSerializer` 
couldn't be converted as it had a constructor argument.

pmck91 committed:
CORE-4806 skip e2etests locally if cluster unavailable (#1396)

Skip e2e tests running locally (not Jenkins) if the K8s cluster is not responding, or not booted up.

chrisr3 committed:
NOTICK: Remove privileged functions from BundleUtils. (#1400)

vlajos committed:
CORE-5260 Merkle: new implementation

josephzunigadaly committed:
CORE-4354: Refactor V1 packaging code to prepare for V2 (#1339)

- Add `CpkReader` and `CpiReader` as entry points for CPK/CPI reading
- Remove `from` method from Cpk and Cpi interfaces
- Expand `FormatVersionReader` to support CPB and CPI format versions
- Add `UnknownFormatVersionException`

chrisr3 committed:
CORE-5052: Upgrade to Corda Gradle plugins 6.0.0-BETA16 and Bnd 6.3.0. (#1409)

Commit:
CORE-2528: Do not include query param if `null` and not required (#1408)

It is HTTP RPC Client change only, as HTTP Server already support not required parameters correctly.
Integration test coverage been extended to expose previously failing scenario.

chrisr3 committed:
NOTICK: Fix publishing for application jars. (#1411)

chrisr3 committed:
NOTICK: Reduce usage of afterEvaluate when publishing applications. (#1412)

williamvigorr3 committed:
CORE-5031 Fix Concurrency issue in delivery tracker (#1404)

Under certain circumstances the DeliveryTracker will replay looped backed messages. This happens if the OutboundMessageProcessor runs quicker the the DeliveryTracker. When the OutboundMessageProcessor processes a message for loopback (i.e. with a Locally Hosted Identity) it adds both a LinkManagerSentMarker and a LinkManagerReceivedMarker to the markers topic.

The DeliveryTracker observers the markers topic. If a LinkManagerSentMarker is observed then the original message is scheduled to replay (based on how long ago the OutboundMessageProcessor sent the original message). If the DeliveryTracker observes a LinkManagerReceived marker then the scheduled replay is canceled. Thus if the DeliveryTracker has a backlog of messages (e.g. because it is slower than the OutboundMessageProcessor) it might cause some messages to replay in the case of looped back messages. As the scheduled message replay might run before the `LinkManagerReceived` marker is processed.

This behaviour was observed when adding the LinkManager to the flow worker process. Adding logging when messages replay, are added and removed from the deliver tracker we see a back log build up and messages start to replay.

I have changed the LinkManager so that loopback messages are not sent if replayed (a simuliar method is used if TTL is expired).

I have generalised the end 2 end test code so we can end 2 end test looped back messages, as well as a unit test of this functionality.

chrisr3 committed:
NOTICK: Publish source jars artifacts, and fix inter-project dependencies for P2P applications.

YashNabar committed:
CORE-5042: Switch to inner classes for membership implementation swaps (#1401)

* CORE-5042: Switch to inner classes for implementation swaps

* Change inactive implementations to private objects

yift-r3 committed:
CORE-4620: Import Certificate API (#1388)

Import Certificates API.

yift-r3 committed:
CORE-5038: Add keys filter APIs (#1410)

Add a few options to filter the keys by an alias, time, category, ids and so on.

yift-r3 committed:
CORE-4833: Membership HSM API (#1406)

Membership HSM API

YashNabar committed:
CORE-4557: Refactor member key properties based on key design (#1403)

The following key properties exposed by the MemberInfo interface have changed -

owningKey changed to sessionInitiationKey
identityKeys changed to ledgerKeys

Relates to corda/corda-api#394

emilybowe committed:
CORE-3776 Add automated testing for invalid requests on the gateway (#1334)

This adds automated testing (a unit test + an integration test) for scenarios where the gateway receives an invalid request.

Sagar-Nenwani-R3 committed:
CORE-4865 Move membership libs and components to use Clock instead of Instant (#1398)

CORE-4865 Move membership libs and components to use Clock instead of Instant
https://r3-cev.atlassian.net/browse/CORE-4865

Files that still use Instant:
* RegistrationRequestProgressDto
* RegistrationRequestProgress
* MemberInfoExtension

vlajos committed:
CORE-5260 Merkle: Transition to digestService

mcgovc committed:
Merge pull request #1414 from corda/chrisr3-publish

NOTICK: Publish source jars artifacts, and fix inter-project dependencies for applications

owenstanford committed:
CORE-5056 Added delegated class loader to allow the kafka client to find the ScramLoginModule class (#1425)

conal-smith committed:
CORE-4868 - added entity version to cpk identifier (#1423)

chrisr3 committed:
CORE-5030: Only remove successfully deleted bundles from their sandbox. (#1407)

davidcurrie committed:
CORE-5060 - Resolve timeouts during topic creation (#1413)

Remove the current resource limits on topic creation as, on a developer
machine, these can double the length of time the job takes to complete.

Also, tweak the startup probes as the workers sometimes fail to start in
the alloted time leading to endless retries. Set the image pull policy
on the topic creation job to ensure we're getting the latest image when
using the unstable tag.

simon-johnson-r3 committed:
CORE-5029 Arm64 host support (#1418)

1. Building on a host which reports 'aarch64' as the platform architecture will result in arm64 Docker images being produced instead of amd64 ones, as long as the base image supports that.
2. Hard coded images and other arch specific items are moved to values so they can be overridden by the developer.

chrisr3 committed:
NOTICK: Prevent anyone applying quasar-utils outside of corda.quasar-app plugin.

mcgovc committed:
Merge pull request #1428 from corda/chrisr3-less-quasar

NOTICK: Prevent anyone applying quasar-utils outside of corda.quasar-app plugin.

pmck91 committed:
CORE-5076 - Changing plugin generation to use Jar file, fixing an issue with plug… (#1309)

* Changing plugin generation to use Jar file, fixing an issue with plugins failing to load on various linux distros

* NOTICK - updating gradle for topic-config

* updating the gradle script to exclude 3rd party manifest files

* Code review changes

chrisr3 committed:
CORE-5081: Upgrade to Kotlin 1.7.0. (#1426)

vlajos committed:
CORE-5260 Merkle: Use normal classes for the specialized nonce cases

vlajos committed:
CORE-5260 Merkle: make depth private for now

E1P3 committed:
CORE-5071 Added Snyk plugin to the build (#1424)

* Add snyk plugin to build settings and properties.gradle

alexey-kadyrov-r3 committed:
CORE-648 fixes based on experience implementing an HSM  (#1422)

vlajos committed:
CORE-5001 mark old party code for future decision

vlajos committed:
CORE-5089 Consensual IOU Mint flow - very WIP

lankydan committed:
CORE-4998 Cleanup flow mapper in-memory state (#1419)

Clean up flow mapper in-memory state of flow starts and sessions.

When a flow finishes of fails, clean up its state.

When a session becomes `CLOSED` or `ERROR`ed, clean up its state.

State is cleaned up by sending a `ScheduleCleanup` record to the flow
mapper, which cleans up the in-memory state after a period of time.

`FlowCheckpointImpl` now _soft deletes_ the Avro `Checkpoint` so that
sessions can still be accessed after marking the `Checkpoint` for
deletion. This is needed if a session goes to `CLOSED` and the flow
finishes all off of a single event, otherwise the `ScheduleCleanup`
record cannot be sent for the session.

This commit also changes the flow mapper's key solely to a `String`. The
string still represents a `FlowKey` (for flow starts) or a session id
(for a session). The change calls `toString` on `FlowKey` rather than
passing in the `FlowKey` directly and letting the state and event
pattern do the conversion to a `String`. This was required to fix a bug
where the partition a state/event was read from was not the same
partition the new state was written to, which indicated a change in the
generated hash of the key. Calling `toString` made the output
consistent and preventing changing the partition.

vlajos committed:
CORE-5001 PartyImpl

JamesHR3 committed:
CORE-4743: Startable by RPC API redesign (#1402)

* CORE-4743: Start integrating the new StartableByRPC API

* CORE-4743: Update the flow processor to handle new RPC startable flows

* CORE-4743: Update to better encapsulate arguments for RPC started flows

* CORE-4743: Also update the initiated flow API to enforce a session input and no output

* CORE-4743: Remove inheritance of flow from rpc startable and responder flows

* CORE-4743: Hide the RPC request args in an object to ensure that the request string is not serialized by default in checkpoints

* CORE-4743: Add a direct JSON conversion method to the RPC request data

* CORE-4743: Update ExampleFlow to use getRequestBodyAs

* CORE-4743: Add a toString method to RPCRequestDataImpl

josephzunigadaly committed:
CORE-4354: Close streams (#1417)

CORE-4354: Close streams

The Windows build detected one of these streams was left open. I went through usages and found a couple more cases.

Looking through this code, I found that CPKs are long lived objects. They held open a file for their lifetime. I've reworked this so they store the filename instead, opening and closing the file only when they want to read from it (in `getResourceAsStream`). As a result the `Cpi` and `Cpk` interfaces no longer need `AutoCloseable`. So you will see some `close` calls removed by this PR.

`CpiLoader` would filter CPKs, potentially losing the last reference without closing (found when they still needed closing). Inspection of the code leading up to this point showed that the filter would never remove CPKs so the filter call was removed along with an unnecessary use of nullable.

yift-r3 committed:
NOTICKET: Fix P2P deployment (#1430)

Fixing two issues with the P2P deployment:

* The publisher assumes that it is in an OSGi context. This change will make it work regardless.
* The configuration version had changed to an integer number. This change will make it version 1 instead of 1.0.

Commit:
CORE-4248: Remove extra quotes in `String` responses (#1438)

Also update integration tests as necessary.

chrisr3 committed:
CORE-5108: Unregister ConfigurationReadService when closing SecurityConfigHandler. (#1434)

davidcurrie committed:
CORE-4686 Make bootstrap resources configurable (#1441)

kyriathar-r3 committed:
CORE-5040 Config DB/Kafka reconciliation integration (#1405) (#1405)

* Move defaulting configurations to `ConfigPublishServiceImpl`, be used by both `ConfigWriterProcessor` and DB config reconciler writer.

* Adds Config reconciler to run once at start up, to offer configurations to `ConfigurationReadService`.

* Removes Reconciliation and RPC boot configurations. These are now added to the DB and are getting reconciled the first time Config reconciliation runs.

* Aligns uses of `Configuration.version` that got changed to an int.

vlajos committed:
CORE-5260 Merkle: use Corda5 secureRandom service

vlajos committed:
CORE-5110 Merkle: SecureHash serialization

Commit:
Fix for BUS config merge.

Commit:
Revert "Fix for BUS config merge."

This reverts commit 77b814d10e9e260f0ab1119aa521eaa83c667d3e.

chrisr3 committed:
NOTICK: Ensure SandboxGroupContextComponent deactivates cleanly. (#1444)

vlajos committed:
CORE-5012 Merkle: Guarantee that the whole tree uses the same digest algorithm

mbrkic-r3 committed:
CORE-4360 Packaging verification library (#1392)

benmcmahon100 committed:
CORE-4597: Add optional topic deletion (#1443)

* Add optional topic deletion
* Copy changes to resources requests and image pull pol
* PR Feedback
* Update charts/corda/values.yaml
* Remove JVM opts

davidcurrie committed:
CORE-5131 Don't delete bootstrap jobs (#1450)

To aid problem determination, don't delete the bootstrap jobs as, at least for now, successful completion of the job doesn't necessarily correspond to success.

Commit:
NOTICK - Helm value files with sensible defaults for local development. (#1448)

* Helm value files with sensible defaults for local development.

vlajos committed:
CORE-5260 Merkle: Suppress Detekt complexity complaints

alexey-kadyrov-r3 commented on libs/crypto/merkle/build.gradle:
```diff
@@ -0,0 +1,22 @@
+    implementation platform("net.corda:corda-api:$cordaApiVersion")
+
+    implementation project(":libs:crypto:crypto-core")
+    implementation project(":libs:crypto:crypto-impl")
```
You don't need that one here

vlajos replied:
Thanks. I have changed it to testImplementation. https://github.com/corda/corda-runtime-os/compare/e21b53bf02777c2afde0ec2495ab5ac8b9eedcc8..a449762cf98ff8b9c7baf3d34432048994cee237

mbrkic-r3 committed:
CORE-4360 Fixed issue with location of GroupPolicy.json (#1458)

Commit:
Patch changes from https://github.com/corda/corda-runtime-os/pull/1453 onto main. (#1463)

RPC RBAC changes:
PermissionManagementCache and PermissionValidationCache re-created every time when config changes. There are other listeners of the config changes which assume that said caches been already created. This may not be always the case as config listeners are being notified in the arbitrary order.
To rectify the issue, said caches are wrapped into AtomicReference allowing them to be initialized at some point in the future.

Lifecycle changes:
Various components updated to close registrations before coordinators as a workaround for an issue in the lifecycle code where doing the reverse results in exceptions. A few subscriptions needed to be closed instead of stopped when they are no longer in use. Finally, a race condition in the lifecycle code can cause registry failures on closing objects; this is rectified by changing the error to a warning.

dimosr committed:
NOTICK - Add remaining changes needed so that p2p can work with static networks (#1433)

A few changes needed to make static networks work with the p2p layer.
* Fixed the topic dumper tool that had been broken by some recent refactoring on the messaging library.
* Adjusted the static network registration handler to populate the hosted identities topic with dummy TLS certificates / session initiations keys (instead of an empty string), since these do not really exist for a static network. Otherwise, the p2p components will fail trying to read the entries. In the future, we can consider making these fields optional, but let's fix things for now and think about that separately.

Commit:
Increase timeout on the ready endpoint test as it may take a while for the cluster to settle after config publication. (#1464)

vlajos committed:
CORE-5109 Merkle: salvage v4 unit tests - first

kyriathar-r3 committed:
CORE-4890 Publish Cpi Info version to Kafka after consulting the DB (#1452) (#1452)

* Uses `CpiMetadataEntity.entityVersion` to populate Cpi Metadata Avro object's entity version

chrisr3 committed:
CORE-5146: Upgrade to Bnd 6.3.1 and Corda Gradle plugins 6.0.0-BETA17. (#1466)

conal-smith committed:
CORE-4868 - update to latest corda api version with cpk_metadata PKs (#1429)

Commit:
CORE-3856 - kafka max allowed msg size config (#1437)

* Set Max Payload size based on maxAllowedMessageSize configuration.
* Choose chunk size based on MaxAllowedMessageSize configuration setting.

JamesHR3 committed:
NOTICK: Set the lifecycle to UP for a few flow worker components (#1465)

vlajos committed:
CORE-5119 Merkle: commenting

Commit:
CORE-2752: Javalin version update to `4.6.3` (#1446)

Also updated SwaggerUI to `4.10.3` and Jetty to `9.4.46.v20220331`.

There were breaking changes during Javalin update:
1. Error messages from JSON parsing have changed;
2. Representation of path parameters have changed from `:pathParam` to `{pathParam}`;
3. `JavalinJackson.configure()` static method removed;
4. `addStaticFiles` has to be told explicitly to load from class path.
Actual static files loading is happening at `io.javalin.Javalin.start()` invocation, hence classloaders has to be setup such that SwaggerUI resource are accessible via `contextClassLoader`.

simon-johnson-r3 committed:
CORE-5107 - Create a demo CorDapp to show off the available functionality in Corda 5 (#1436)

Also fixes missing X500 Principal kryo serialization support. Without this the chat app cannot determine the message sender.

barrylapthorn committed:
CORE-3989 vnode reconciliation (#1454)

We need a 'vnode-info writer' for the reconciler to use.
Previously it was hidden inside another component.

Move virtual node 'services' to their own component to reduce
complexity in the db-processor component and remove some
detekt issues regarding long parameter lists and function
complexity.

Add test to check we can persist the holding id, and then
persist the vnode.

Extend db entity to include required fields and timestamp.

Introduce audit columns

Use hibernate properly and join tables by _objects_
not by _id_.

Ensure that we're not lazily fetching results by testing
values outside a transaction block.

vlajos committed:
CORE-5109 Merkle: Do not create proof for duplicated indices

vlajos committed:
CORE-5109 Merkle: Port not-yet covered case's tests from old code and comment a bit the tests

Commit:
CORE-4881 - DBChange from database source (#1471)

* Introduce an implementation of DbChange that supports DB Change Log files loaded from the DB (through JPA entities).
Integration will happen in a follow-up change.

davidcurrie committed:
CORE-5192 Specify replicas for Kafka/ZK (#1472)

williamvigorr3 committed:
CORE-4437 Clean up Domino Logic (#1331)

Clean up the Domino Logic:

- Refactor ComplexDominoTile renaming createResources to onStart and removing the resource holder. Refactored the Link Manager so resources are created outside of the tile. 

- Make DominoTileState completely internal to each DominoTile. Currently LifecycleStatus is exposed instead but this can probably be made internal to the tile (although might be useful for unit testing).

mbrkic-r3 committed:
CORE-4360 Packaging verification library - CPx format 2 (#1451)

JamesHR3 committed:
CORE-5100: Retrieve the correct sandbox in the FlowFiberExecutionContextFactoryImpl (#1476)

Commit:
Improve checking of status endpoint in case status goes up/down/up. (#1474)

JamesHR3 committed:
CORE-5102: Flow marker interface and generic marshalling service (#1447)

* CORE-5102: Add a generic marshalling service and move the JSON marshalling service to use it

* CORE-5102: Introduce common Flow interface and move old Flow to Subflow

* CORE-5102: Add further test and update the CorDapp manifest

* CORE-5102: Add more tests for the new flow implementation classes

* CORE-5102: Move the parseList API down to the MarshallingService

* CORE-5102: Add getRequestBodyAsList method to RPCRequestData

JasonSinclair95 committed:
Core 4600 kafka auth attempt2 (#1439)

* changes made for kafka authentication

simon-johnson-r3 committed:
CORE-5166 Improve logging in the avro serializer (#1477)

kyriathar-r3 committed:
NOTICK Update `VersionedRecord` KDoc (#1486)

chrisr3 committed:
CORE-5246: Replace Aries dynamic bundle with Aries dynamic framework extension. (#1484)

chrisr3 committed:
NOTICK: Repair breakage of JSON message parsing. (#1485)

dimosr committed:
CORE-3153 - Add integration with crypto ops client in the gateway (#1480)

Adding integration with the `CryptoOpsClient` in the gateway to delegate TLS signing to. The existing integration with our stub crypto processor is kept in place so that end-to-end testing can be performed in our existing setup and will be removed once integration with MGM/dynamic networks has been completed and tested succesfully. An option has been added to switch between stubs or the real crypto processor in the Gateway worker. The default has been set up to use the stubs, so that our end-to-end tests continue working without further changes needed.

yift-r3 committed:
NO-TICKET: Make category in keys API case insensetive (#1442)

vlajos committed:
CORE-4996 Restructure repository for ledger works (#1468)

Rename existing ledger module in the API to `ledger-obsolete` so we can free the package namespace for future ledger implementations.

conal-smith committed:
CORE-4868 - correct version handling for merging CPK entities (#1483)

before merging CPI and CPK entities we must find the entities first by their primary keys and if the entities exist, update these before merging rather than creating a new entity each time with the default entityVersion = 0
